crates/wasmtime/src/config.rs

```diff
@@ -330,6 +330,23 @@ impl Default for InstanceAllocationStrategy {
     }
 }
 
+#[derive(Clone)]
+/// Configure the strategy used for versioning in serializing and deserializing [`crate::Module`].
+pub enum ModuleVersionStrategy {
+    /// Use the wasmtime crate's Cargo package version.
+    CargoPkgVersion,
```

Comment:
🚲 shed: perhaps WasmtimeVersion to remove "cargo" from the identifier if we want to use this 1:1 from other language bindings (they don't know what cargo is).

Reply:
Thanks, I like this idea.

The other thing to figure out is whether to sanitize custom strings to be less than 256 bytes somehow, or to change the length field in the binary to be a u64 instead of a byte.

Reply:
I don't think it's unreasonable to document a length limit, personally.

We have other settings that return [`Result<&mut Self>`](https://github.com/bytecodealliance/wasmtime/blob/main/crates/wasmtime/src/config.rs#L587) too, so we could verify it at the `Config` level.